index.js

```diff
@@ -27,24 +29,55 @@ try {
  * @method resolvePackagePathSync
  * @param {string} name name of the dependency module.
  * @param {string} basedir root dir to run the resolve from
- * @param {Boolean} shouldCache (optional) intended to bypass cache defaulting to true
+ * @param {Boolean|CustomCache} (optional)
+ *  * if true: will choose the default global cache
+ *  * if false: will not cache
+ *  * if undefined or omitted, will choose the default global cache
+ *  * otherwise we assume the argument is an external cache of the form:
+ *      cache = {
```

Comment:
This doesn't match the definition you have above:
cache = ``{
  RESOLVED_PACKAGE_PATH: new Map(),
  REAL_FILE_PATH: new Map(),
  REAL_DIRECTORY_PATH: new Map()
};

Reply:
(Or, because we're in a different file, is my comment irrelevant?)